src/include/sof/math/trig.h

```diff
@@ -21,17 +22,38 @@
 #define PI_Q4_28      843314857
 #define PI_MUL2_Q4_28     1686629713
 #define CORDIC_31B_TABLE_SIZE		31
-#define CORDIC_SIN_COS_15B_TABLE_SIZE	15
+#define CORDIC_15B_TABLE_SIZE		15
+#define CORDIC_30B_ITABLE_SIZE		30
+#define CORDIC_15B_ITABLE_SIZE		15
 
 typedef enum {
 	EN_32B_CORDIC_SINE,
 	EN_32B_CORDIC_COSINE,
+	EN_32B_CORDIC_CEXP,
 	EN_16B_CORDIC_SINE,
 	EN_16B_CORDIC_COSINE,
+	EN_16B_CORDIC_CEXP,
+	EN_32B_CORDIC_ASINE,
+	EN_32B_CORDIC_ACOSINE,
 } cordic_cfg;
+struct cmpx {
+	int32_t re;
+	int32_t im;
+};
 
-void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t *b_yn, int32_t *xn,
-		    int32_t *th_cdc_fxp);
+ /* calculate LUT = 2*atan(2.^(-1:-1:(nIters)+1)) */
+static const int32_t cordic_ilookup[CORDIC_30B_ITABLE_SIZE] = {
```

Comment:
Does this table get instantiated to every module that includes trig.h? If so, it consumes lot of RAM. It could be moved to beginning of trig.c.

Reply:
Thanks - Good idea .Preference is for private header